T5 550 generator

mathisschiwy opened the pull request.

mathisschiwy committed:
Basic Project functions with unit tests

mathisschiwy committed:
removed .gitignore from repository

mathisschiwy committed:
changed name

mathisschiwy committed:
Some changes due to better readabilty

mathisschiwy committed:
Improvements and yaml file

JessicaWu1 commented on project_order_dice/tests/test_enter_names.py:
```diff
@@ -0,0 +1,43 @@
+class TestInputFunction(unittest.TestCase):
+    def test_enter_names(self):
+        names = ["Hannes", "Jessica", "Henning", "Mathis", "Sarah", "Steffen"]
+        mock_input = ["Hannes", "Jessica", "Henning", "Charlie", ""]
```
Charlie is in the test and it therefore fails because it can not be removed

JessicaWu1 commented on project_order_dice/tests/test_enter_names.py:
```diff
@@ -0,0 +1,43 @@
+
+            self.assertEqual(names, expected_names)
+
+    def test_exclude_enter_names(self):
```
This test does not seem to run for me, the exception in names.remove() is still thrown, which leads to an error and the test does not continue past that point.

JessicaWu1 commented on project_order_dice/main.py:
```diff
@@ -1,14 +1,13 @@
+print("If all members are present simply press return")
+
+# Function to exclude the names of people who are not present.
+functions.exclude_enter_names(names)
```
Do we still need to call the exclude_enter_names function if the YAML file is read and used?
Maybe if we toggle the YAML file usage then it would be useful (either use enter names or yaml file?)

JessicaWu1 commented on project_order_dice/functions.py:
```diff
@@ -0,0 +1,38 @@
+                names.remove(not_present)
+            elif not_present == "":
+                break
+        except:
```
You are not defining which exception should be catched. This should be a ValueError.

mathisschiwy committed:
Changed tests

Commit:
Update project order dice config: Henning is back

mathisschiwy committed:
Added a featuere to configure days on which people are not available

mathisschiwy committed:
Merge remote-tracking branch 'origin/T5-550_generator' into T5-550_generator

mathisschiwy committed:
Added Adrain to the oprder dice config

mathisschiwy committed:
Updated unittests

Commit:
Add basic readme

mathisschiwy committed:
Merge pull request #7 from mathisschiwy/T5-550_add_basic_readme

Add basic readme for daily dice

henning commented on project_order_dice/Config/away.yaml:
```diff
@@ -0,0 +1,34 @@
+# If a person is away for a longer period of time please put them on is-away: True.
+away:
```
This list does not only specify who is away, it's also the main input of which team member names we use for the random order. 

Could for example be named just "team members"

henning replied:
Also, because this file then is nopt reduced to define away behaviour, it would better be named just "config.yaml" in my opinion. 

Because then we have a file config.yaml in a directory config, without any other config files for now, it would be easier to remove the directory and have the config file in the main directory.

henning commented on project_order_dice/README.md:
```diff
@@ -0,0 +1,12 @@
+
+# Usage:
+
+TODO
```
Here we want to explain the usage before we merge it (e.g. just do ./main.py)

henning commented on project_order_dice/README.md:
```diff
@@ -0,0 +1,12 @@
+* you need to have pyyaml installed - it's tested it with version 5.4.1, 
+  others might or might not work
+  * see https://pypi.org/project/PyYAML/5.4.1/
+
```
maybe add python3 as requirement. (some systems and people still have just python2 - lets help them to know with python 2 they cannot run the program)

henning commented on project_order_dice/functions.py:
```diff
@@ -0,0 +1,87 @@
+        weekday = today.isoweekday()
+
+    # Check weekdays
+    try:
```
Maybe we could think about to keep the code a bit more on the left (less cascaded if if try blocks) - but I'm not sure if it's useful and possible in the concrete case

henning reviewed (changes requested):
I'd like to see the readme be finished and the config file renamed, then I want top approve it finally! thanks ;)

mathisschiwy committed:
Added some changes on the yaml file and Readme

mathisschiwy committed:
Added Python badge to Readme

henning commented on project_order_dice/functions.py:
```diff
@@ -0,0 +1,87 @@
+    try:
+        if weekday == 1:
+            monday_names = data.get("Monday", [])
+            for names in monday_names:
```
Regarding my wish to do less stuff in nested blocks indented deeply to the right: 
the "for names in DAYNAME_names" does not need to be repeated for each day, 
If you store the names for the current day in a name-neutral variable (e.g. "today_names", then you need to have the filtering code only once.

henning commented on project_order_dice/functions.py:
```diff
@@ -0,0 +1,87 @@
+            friday_names = data.get("Friday", [])
+            for names in friday_names:
+                filtered_names = [name for name in filtered_names if name not in names.get("names", [])]
+    except TypeError:
```
Catching exceptions and just passing them is most often not the best practice, and it should at least be commented why one does it if one does it.

Like now, I would like to try to think about a way that we dont even need the try/catch block and avoid the exception happening in first place, but I dont even understand why it is there in the first place.

Commit:
Proposal to make some things more readable

* for loop instead of list comprehension - also eliminates the need of
  exception handling
* one loop/list comprehension can be done only once to get list
  relevant for today

mathisschiwy committed:
Refactor: handle empty entries, improve readability

We had errors when vacation entries were not properly filled and readability of list comprehension was less then ideal, so we replaced it with a for loop

mathisschiwy committed:
Merge pull request #8 from mathisschiwy/function_readability_proposal

Proposal to make some things more readable

mathisschiwy committed:
Added mock.yaml and adjust unittest

mathisschiwy committed:
Added a few comments

henning commented on project_order_dice/main.py:
```diff
@@ -1,14 +1,16 @@
-for name in random.sample(names, len(names)):
-    print(name)
+# Function to manually exclude people who are not available at the current day
+functions.manuel_exclude(filtered_names)
```
small typo: it's "manual" with a

mathisschiwy replied:
Improved it

mathisschiwy committed:
Improved a small typo

henning commented on project_order_dice/tests/mock.yaml:
```diff
@@ -0,0 +1,34 @@
+# If a person is away for a longer period of time please put them on is-away: True.
```
please use the name text fixture for test fixtures ;)

mathisschiwy committed:
Refactore mock.yaml to test_fixture.yaml

henning reviewed (dismissed):
Looks good

henning reviewed (approved):
Looks good
